### DISCUSSION TITLE
Discussion For New Feature Requests

### DISCUSSION THREAD

**Discussion by anant-svc** [Greg0/Lazer-Database#68] [Ideas]
Hello,

I hope you all are doing good to great.

It being almost 1 year that there is not new features included.

I am opening this discussion for requesting new features to include in next releases of Lazer Database.

You can share your thoughts and suggestions here.

**Comment by woody180** [comment]
I think it must be great if there would been feature that automatically removes shared dependencies. For example if I have an article that belongs to some category with junction table (many to many relationships). If I remove some category or article, it will be nice records to remove from junction table automatically.

Example:


        // 1. Fetch record to be removed
        $tobeRemoved = db::table('articles')->with('articlecategories')->where('id', '=', $id)->findAll();

        // 2. Delete article 
        db::table('articles')->find(reset($tobeRemoved)[0]->id)->delete();

        // 3. Get junction table
        $junction = Relation::table('articles')->with('articlecategories')->getJunction();

        // 4. Delete related junction records from junction table
        foreach ($tobeRemoved as $rec)
            foreach ($rec->Articlecategories->findAll() as $related)
                db::table($junction)->where('articles_id', '=', $rec->id)->andWhere('articlecategories_id', '=', $related->id)->find()->delete();

**Reply by anant-svc** [reply]
@woody180 I have read your feature request. You may know Lazer-Database already has Relation with Many to Many and 2 others.
I think we didn't prepared PHP Unit test for your scenario so I will prepare it and will get back to you with full detail.
URL: https://github.com/Lazer-Database/Lazer-Database#relations

**Comment by anant-svc** [comment]
Do you want to drop constant LAZER_DATA_PATH support? I can provide you a patch files then you can use multiple database in single process.

**Comment by ghost** [comment]
Add jsonpath in Lazer-Database?

> Lazer::table('table_name')->where('name', '=', 'John')->find()->delete(); //Will remove John from DB


vs

$.table_name.name["John"].delete()

?

**Reply by anant-svc** [reply]
Welcome @stokebreakup, I appreciate your idea to shorten by jsonpath and simple CRUD operations.
It is not possible in the core though I will try to create a separate file to extend it.

Here is tool to migrate any data to Lazaer Database format, you don't have to map any fields/columns.
This tool will create tables according to the value automatically and store data for you.
Have a look here: https://github.com/Lazer-Database/Migrate-to-Lazer-Database

**Reply by anant-svc** [reply]
@stokebreakup spread your love by hitting Star in https://github.com/Lazer-Database/Migrate-to-Lazer-Database

**Comment by anant-svc** [comment]
Open to Ideas for the new release

**Comment by anant-svc** [comment]
@woody180 I have launched new docs website - [lazer-database.pages.dev/](https://lazer-database.pages.dev/)

**Reply by anant-svc** [reply]
Doc website is back.

**Comment by anant-svc** [comment]
Today I have updated the GitHub workflow to test PHP 8.2, 8.3, 8.4 and 8.5. All tests are successful and in future we can improve the code.
